Source/DafnyLanguageServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -239,25 +239,9 @@ public class DafnyModel {
     /// <summary>
     /// Return the name of a 0-arity type function that maps to the element if such
     /// a function exists and is unique. Return null otherwise.
-    /// If the name is also aliased by a type parameter, return the name of the concrete type. 
     /// </summary>
     private static string GetTrueTypeName(Model.Element element) {
-      string name = null;
```

Comment:
What's happening  with the test `(name == null) || name.Contains("$") || name.StartsWith("#") || name.Contains("@"`? Why is it no longer necessary?

Reply:
Good question. I'll have to look at it more carefully. I'm not sure off the top of my head.

Reply:
I think it is no longer possible for `name` to have these special characters after monomorphization.  I remember asking @Dargones to review the changes in this part of the PR as they affect test generation, and if I recall correctly he mentioned that he tested them on his branch and were ok.